feat(dashboards): Implement `BigNumberWidget` Badge feature

gggritso opened the pull request:
Contribute to https://github.com/getsentry/sentry/issues/77779

**e.g.,**
![Screenshot 2024-10-10 at 12 02 06 PM](https://github.com/user-attachments/assets/4428ea6e-0b6f-46e7-a80d-dfc7797d45b1)

gggritso committed:
Give props a more descriptive name

gggritso committed:
Implement badge feature

narsaynorath commented on static/app/views/dashboards/widgets/common/widgetFrame.stories.tsx:
```diff
@@ -71,6 +71,33 @@ export default storyBook(WidgetFrame, story => {
+          <NormalWidget>
+            <WidgetFrame
+              title="count()"
+              badge={{
```
Would the API be clearer if we called it `badgeProps` instead of `badge`?

gggritso replied:
I like this idea! Changed

narsaynorath reviewed (approved):
small comment, lgtm

gggritso committed:
Merge branch 'master' into feat/dashboards/widget-frame-badge

gggritso committed:
Rename prop

`badgeProps` is clearer, and is an established pattern. We often do
`XProps` if the props are relevant so a sub-component only.
